src/engine/contactwriter.cpp

```diff
@@ -4745,9 +4803,14 @@ QContactManager::Error ContactWriter::syncUpdate(const QString &syncTarget,
                     QMap<QContactDetail::DetailType, DetailPair>::const_iterator cit = composedModifications.constBegin(), cend = composedModifications.constEnd();
                     for ( ; cit != cend; ++cit) {
                         // Apply these modifications to empty details, and add to the contact
+                        // XXX TODO: WHY? What is a "composed modification" ?
```

Comment:
Composed modification means a modification to a detail type that is subject to composition, meaning it derives from details which are stored in the primary contact table rather than in a typed detail table; for example: contact name.  These details are different in that the resulting aggregate value cannot be accurately traced back to its origin when multiple constituents are aggregated together, unlike the details which are not subject to composition.